File: src/main/java/com/kiseru/asteroids/client/Receiver.java
package com.kiseru.asteroids.client;

import java.io.BufferedReader;
import java.io.IOException;

public class Receiver implements Runnable {
    private BufferedReader reader;

    public Receiver(BufferedReader reader) {
        this.reader = reader;
    }

    @Override
    public void run() {
        while (true) {
            try {
                String inputData = reader.readLine();
                System.out.println(inputData);
            } catch (IOException e) {
                throw new RuntimeException(e);
            }
        }
    }
}

File: src/main/kotlin/com/kiseru/asteroids/client/Receiver.kt
package com.kiseru.asteroids.client

import java.io.BufferedReader
import java.io.IOException

class Receiver(private val reader: BufferedReader) : Runnable {

    override fun run() {
        while (true) {
            try {
                val inputData = reader.readLine()
                println(inputData)
            } catch (e: IOException) {
                throw RuntimeException(e)
            }
        }
    }
}
